sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/xml/XmlDataSource.scala

```diff
@@ -174,15 +175,26 @@ object MultiLineXmlDataSource extends XmlDataSource {
       file: PartitionedFile,
       parser: StaxXmlParser,
       requiredSchema: StructType): Iterator[InternalRow] = {
-    parser.parseStream(
-      CodecStreams.createInputStreamWithCloseResource(conf, file.toPath),
-      requiredSchema)
+    if (SQLConf.get.legacyXMLParserEnabled) {
```

Comment:
can we define a CSV option for it? A similar example in `CSVOptions`:
```
  val inputBufferSize: Option[Int] = parameters.get(INPUT_BUFFER_SIZE).map(_.toInt)
    .orElse(SQLConf.get.getConf(SQLConf.CSV_INPUT_BUFFER_SIZE))
```

This allows users to turn on/off the legacy parser for each query, instead of the entire session.

Reply:
Done, please take a look!